### PR TITLE
Task/bws80 restriction to add worker to the group of another tenant

### PR DESCRIPTION
**Pull request by Oleg-feriss**

### CODE REVIEW

File: backend/src/services/worker/worker.service.ts [hunk 1]
```diff
@@ -104,12 +104,25 @@ class Worker {
...
+    const groupIdsByTenant = await this.#workerRepository.getGroupIdsByTenant(
+      groupIds,
+      master.tenantId,
+    );
+
+    if (groupIdsByTenant.length === 0) {
+      throw new InvalidCredentialsError({
+        status: HttpCode.DENIED,
+        message:
+          'That group does not exist. Please select any group or create a new one first',
```
Comment (what1s1ove) [inline backend/src/services/worker/worker.service.ts]: use enum

File: backend/src/services/worker/worker.service.ts [hunk 2]
```diff
@@ -104,12 +104,25 @@ class Worker {
       passwordSalt,
     );
 
+    const groupIdsByTenant = await this.#workerRepository.getGroupIdsByTenant(
+      groupIds,
+      master.tenantId,
+    );
+
+    if (groupIdsByTenant.length === 0) {
+      throw new InvalidCredentialsError({
```
Comment (what1s1ove) [inline backend/src/services/worker/worker.service.ts]: in isn't a credentialing error

File: backend/src/services/worker/worker.service.ts [hunk 3]
```diff
@@ -104,12 +104,25 @@ class Worker {
       passwordSalt,
     );
 
+    const groupIdsByTenant = await this.#workerRepository.getGroupIdsByTenant(
+      groupIds,
+      master.tenantId,
+    );
+
+    if (groupIdsByTenant.length === 0) {
```
Comment (what1s1ove) [inline backend/src/services/worker/worker.service.ts]:
```suggestion
    if (hasGroups) {
```

**Review by what1s1ove — CHANGES_REQUESTED** [review]

File: backend/src/data/repositories/worker/worker.repository.ts [hunk 4]
```diff
@@ -86,6 +90,26 @@ class Worker {
...
+  ): Promise<string[]> {
+    const groups = await this.#GroupModel
+      .query()
+      .select('id')
+      .where({ tenantId });
+
+    const groupIdsByTenant = groups
+      .map((group) => group.id)
+      .filter((id) => {
+        for (const groupId of groupIds) {
```
Comment (what1s1ove) [inline backend/src/data/repositories/worker/worker.repository.ts]: use `array.some`

**Review by what1s1ove — CHANGES_REQUESTED** [review]

File: backend/src/data/repositories/worker/worker.repository.ts [hunk 5]
```diff
@@ -86,6 +90,24 @@ class Worker {
...
+  ): Promise<string[]> {
+    const groups = await this.#GroupModel
+      .query()
+      .select('id')
+      .where({ tenantId });
+
+    const groupIdsByTenant = groups
+      .map((group) => group.id)
+      .filter((id) => {
+        groupIds.some((groupId) => groupId === id);
```
Comment (what1s1ove) [inline backend/src/data/repositories/worker/worker.repository.ts]: just some? return?

File: backend/src/services/worker/worker.service.ts [hunk 6]
```diff
@@ -104,12 +106,23 @@ class Worker {
       passwordSalt,
     );
 
+    const groupIdsByTenant = await this.#workerRepository.getGroupIdsByTenant(
+      groupIds,
+      master.tenantId,
+    );
+
+    const hasGroups = groupIdsByTenant.length > 0;
```
Comment (what1s1ove) [inline backend/src/services/worker/worker.service.ts]:
```suggestion
    const hasGroups = Boolean(groupIdsByTenant.length);
```

**Review by what1s1ove — CHANGES_REQUESTED** [review]

File: backend/src/data/repositories/worker/worker.repository.ts [hunk 7]
```diff
@@ -86,6 +90,24 @@ class Worker {
...
+      .where({ tenantId });
+
+    const groupIdsByTenant = groups
+      .map((group) => group.id)
+      .filter((id) => {
+        return groupIds.some((groupId) => groupId === id);
+      });
+
+    return groupIdsByTenant;
+  }
```
Comment (Yaroslaww-1) [inline backend/src/data/repositories/worker/worker.repository.ts:109]: What does this method should do? Why `groupsIds` param is passed here? Remember, repository is only for data aceess. If you need to postprocess/postfilter data It's better to move this logic outside of repository.

**Review by Yaroslaww-1 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**